src/Illuminate/Container/Container.php

```diff
@@ -196,7 +196,7 @@ public function resolved($abstract)
     public function isShared($abstract)
     {
         return isset($this->instances[$abstract]) ||
-               (isset($this->bindings[$abstract]['shared']) &&
```

Comment:
This change looks wrong.

Reply:
Because the 'shared' key always exist as long as the corresponding `$abstract` exists. I think this would be enough:

```php
(isset($this->bindings[$abstract]) && $this->bindings[$abstract]['shared'] === true)
```
---

Just like what we did with the `getConcrete()` method: https://github.com/laravel/framework/blob/6.x/src/Illuminate/Container/Container.php#L729

```php
if (isset($this->bindings[$abstract])) {
  return $this->bindings[$abstract]['concrete'];
}
```
In this case, we also know for sure that the `concrete` key is always availabel as long as the `$abstract` key exists.